Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -107,14 +109,38 @@ internal Initializer(ComputeCredential other)
         public ComputeCredential() : this(new Initializer()) { }
 
         /// <summary>Constructs a new Compute credential instance.</summary>
-        public ComputeCredential(Initializer initializer) : base(initializer) => OidcTokenUrl = initializer.OidcTokenUrl;
+        public ComputeCredential(Initializer initializer) : base(initializer)
+        {
+            OidcTokenUrl = initializer.OidcTokenUrl;
+            if (HasExplicitScopes)
+            {
+                var uriBuilder = new UriBuilder(TokenServerUrl);
+                string scopesQuery = $"scopes={string.Join(",", Scopes)}";
+
+                // As per https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder.query?view=net-6.0#examples
+                if (uriBuilder.Query is null || uriBuilder.Query.Length <= 1)
+                {
+                    uriBuilder.Query = scopesQuery;
+                }
+                else
+                {
+                    uriBuilder.Query = $"{uriBuilder.Query.Substring(1)}&{scopesQuery}";
```

Comment:
So *setting* the query you don't want the leading "?" (prior to .NET 5), but *fetching* it you receive it? That sucks so much.